ext/pcre/php_pcre.c

```diff
@@ -1720,6 +1720,14 @@ PHPAPI zend_string *php_pcre_replace_impl(pcre_cache_entry *pce, zend_string *su
 
 			if (new_len >= alloc_len) {
 				alloc_len = zend_safe_address_guarded(2, new_len, 0);
+				if (UNEXPECTED(alloc_len > ZSTR_MAX_LEN)) {
```

Comment:
This is pretty unfortunate in that the zend_safe_address_guarded above is already there to check for overflow, and then we have another check after that :/

Reply:
Alternatively, we could introduce a `zend_safe_string_alloc()` (or maybe reconsider to check for that overflow in `zend_string_alloc()` only for 32 bit architectures?)

Reply:
I have now added `zend_safe_string_address_guarded()` to avoid the double check.

Reply:
Solved it differently according to Dmitry's comment.